src/librustdoc/doctest.rs

```diff
@@ -672,18 +672,6 @@ fn run_test(
             "--extern=doctest_bundle_{edition}=",
             edition = doctest.edition
         ));
-        for extern_str in &rustdoc_options.extern_strs {
```

Comment:
Also note for additional context: This didn't apply to doctest-only dependencies as registered via `--doctest-compilation-args` as noted here: https://github.com/rust-lang/rust/pull/137899#discussion_r1986463750